modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs

```diff
@@ -323,8 +325,27 @@ private void _ExportBeginImpl(string[] features, bool isDebug, string path, long
                                 {
                                     if (platform == OS.Platforms.Android)
                                     {
-                                        if (IsSharedObject(Path.GetFileName(path)))
+                                        var fileName = Path.GetFileName(path);
+                                        if (IsSharedObject(fileName))
                                         {
+                                            if (fileName.EndsWith(".jar"))
+                                            {
+                                                // Don't export the same dotnet jar twice. Otherwise we will have conflicts. This
+                                                // can happen when exporting for multiple architectures. The jars exported from
+                                                // godot for dotnet can be found in .godot/mono/temp/bin/Export[Debug|Release].
+                                                // If you compare the bytecode with AndroidStudio or IntelliJ, you will find that
+                                                // they are identical. Hence why we can only export one.
+                                                //
+                                                // Gradle does not distinguish jars between abi folders like it does with .so files. It will
+                                                // try to compile both into the application if they are both exported.
+                                                if (exportedJars.Contains(fileName))
+                                                {
+                                                    return;
+                                                }
+
+                                                exportedJars.Add(fileName);
+                                            }
+
```

Comment:
@raulsntos As mentioned in RC, I think we should exclude the `jar` file from being exported via `add_shared_object`.
Instead, we should create a `monoLibs` folder under `platform/android/java/app/`, move `libSystem.Security.Cryptography.Native.Android.jar` there, and update the `build.gradle` dependency to match.

This will slightly increase the size of the standard gradle build template, but I think it's a tradeoff worth having, as it'll allow us to use the same gradle build template for both `standard` and `mono` builds.

Reply:
@m4gr3d or @raulsntos Do either of you know how to implement this?  I personally do not yet see the vision in order to implement it.  Better is better and I like better.  I'm just missing some context on how to change this PR to match your suggestions.

Reply:
What @m4gr3d is saying is that, instead of taking the `jar` file from the .NET build, we can use the one included in the Godot repo here: [modules/mono/thirdparty/libSystem.Security.Cryptography.Native.Android.jar](https://github.com/godotengine/godot/blob/0f20e67d8de83c30b5dd79cb68d12d4cf613065d/modules/mono/thirdparty/libSystem.Security.Cryptography.Native.Android.jar).

You would move that to the suggested directory: `platform/android/java/app/monoLibs`.

Then, update the `build.gradle` to point to that path. This is the part that needs to be changed:

https://github.com/godotengine/godot/blob/0f20e67d8de83c30b5dd79cb68d12d4cf613065d/platform/android/java/app/build.gradle#L72-L76

You'd change the path to be relative to the `platform/android/java/app` directory, so it works in user Gradle builds, so it'd be: `monoLibs/libSystem.Security.Cryptography.Native.Android.jar`. And we don't need to check if the `jar` exists anymore because now it's guaranteed to always be there[^1].

Lastly, change `ExportPlugin.cs` to exclude every `jar` files, since the required `jar` will already be included in the Gradle template instead.

Personally, I'm a bit worried that this could break if a future .NET version changes the `jar` since, with this change, we'll always be including the `jar` from .NET 8 and may result in broken builds if the user project targets a different version that requires a different `jar`. But at least it should fix Gradle builds for 4.4, and we can always improve it later if needed.

[^1]: This check was added originally because we wanted Gradle builds to pull the `jar` from the user project build, instead of including it in the template.

Reply:
@raulsntos brings up a good point regarding targeting different .net versions and future breakage. Is there any reason why we wouldn't want to do it the way it is currently set up in this PR?

Reply:
> https://docs.godotengine.org/en/stable/contributing/development/compiling/compiling_with_dotnet.html

Note that this is the documentation for building the engine. For users we recommend to always use the latest SDK version, but you may still target a different version in your csproj. For example, you can use the .NET 9 SDK to build a C# project that targets .NET 8.

We'll also bump the `GodotSharp` target to `net8.0` in 4.4 which means user projects must target at least .NET 8, but they can always target a newer version (e.g.: .NET 9).

> But still, what version does the current jar target?

It was retrieved from a .NET 8 build, if I remember correctly.

> I'm personally in favor of keeping it the way this PR does it. But I'll change it (or at least add a boolean for users to decide) if anyone of you prefers I do that instead.

Since we're getting close to the 4.4 release, I think it'd be better to change this PR to bundle the `.jar` as described. I think that'd be more likely to get merged quickly, so we can make sure Gradle builds are not broken when 4.4 is released. Then, we can improve upon it in a separate PR, without the pressure of the 4.4 release.

Reply:
I agree.  Here is my suggested plan of action:

1. I will open a new issue regarding retrieving updates to Microsoft distributed jars for .NET.  This PR will solve that issue so I will leave this PR here and open.  Even if we come up with a different approach, this PR can serve as an option for 4.5 until we come up with something better.

2. We add https://github.com/godotengine/godot/issues/98064 to the 4.4 milestone as this issue is a regression and will break android devs.  I do not think we should ship 4.4 with this issue still open.

3. We create a new PR with @m4gr3d 's suggestion above.  I personally cannot commit to doing that PR at this very moment.  My immediate efforts are directed at getting our game ready to start early testing with our users this late fall / early winter.

What do you think?  And is this a PR @m4gr3d or someone else would be able to do in the mean time?  If my time does free up before someone else puts in the PR, I'll gladly do it myself.  But I don't foresee that happening before the 4.4 release.